src/fibers_meta.jl

```diff
@@ -30,6 +30,43 @@ function fiber(arr, default=zero(eltype(arr)))
     Base.copyto!(Fiber((DenseLevel^(ndims(arr)))(Element{default}())), arr)
 end
 
+@generated function Base.:(==)(A::Fiber, B::Fiber)
+    A = virtualize(:A, A, LowerJulia())
+    idxs = [Symbol(:i_, n) for n = getsites(A)]
+    idk = getsites(A)
+    return quote
+        check = Scalar(size(A) == size(B))
```

Comment:
This is a small suggestion: The Julia compiler likely doesn't understand the contents of the `Scalar` type very well, so it may be better to implement this with the single line `size(A) == size(B) || return false` at the top, then the rest of the function below. If the sizes are known statically, Julia can then constant propagate and inline the function to false.